R/utils.R

```diff
@@ -7,25 +7,25 @@
 #' @param pattern string to search in the forecaster name.
 #'
 #' @export
-forecaster_lookup <- function(pattern, forecaster_grid = NULL, printing = TRUE) {
-  if (is.null(forecaster_grid)) {
-    cli::cli_warn("Reading `forecaster_param_combinations` target. If it's not up to date, results will be off. Update with `tar_make(forecaster_parameter_combinations)`.")
-    forecaster_grid <- tar_read_raw("forecaster_parameter_combinations") %>%
-      map(make_forecaster_grid) %>%
-      bind_rows()
+forecaster_lookup <- function(pattern, forecaster_parameter_combinations = NULL) {
+  if (is.null(forecaster_parameter_combinations)) {
+    cli::cli_warn("Reading `forecaster_param_combinations` target. If it's not up to date, results will be off.
+    Update with `tar_make(forecaster_parameter_combinations)`.")
+    forecaster_parameter_combinations <- tar_read_raw("forecaster_parameter_combinations")
   }
-  fc_row <- forecaster_grid %>% filter(grepl(pattern, id))
-  if (printing) {
-    params <- fc_row$params[[1]]
-    if (!is.null(params$trainer)) {
-      params$trainer <- as_string(params$trainer)
+
+  # Remove the "forecaster_" prefix from the pattern if it exists.
+  if (grepl("forecast_", pattern)) {
+    pattern <- gsub("forecaster_", "", pattern)
+  }
+
+  for (table in forecaster_parameter_combinations) {
+    filtered_table <- table %>% filter(grepl(pattern, id))
+    if (nrow(filtered_table) > 0) {
+      filtered_table %>% glimpse()
+      break
     }
-    print(glue::glue("name: {fc_row %>% pull(id)}"))
-    print(glue::glue("forecaster: {fc_row$forecaster[[1]]}"))
-    print(glue::glue("params:"))
-    print(params %>% data.table::as.data.table())
   }
-  return(fc_row)
```

Comment:
if glimpse isn't showing everything it'd be nice to have the result to actually look at